src/native/managed/cdacreader/src/Data/PEAssembly.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Diagnostics.DataContractReader.Data;
+
+internal sealed class PEAssembly : IData<PEAssembly>
+{
+    static PEAssembly IData<PEAssembly>.Create(Target target, TargetPointer address)
+        => new PEAssembly(target, address);
+
+    public PEAssembly(Target target, TargetPointer address)
+    {
+        Target.TypeInfo type = target.GetTypeInfo(DataType.PEAssembly);
+
+        TargetPointer peImagePointer = target.ReadPointer(address + (ulong)type.Fields[nameof(PEImage)].Offset);
+        if (peImagePointer != TargetPointer.Null)
+            PEImage = target.ProcessedData.GetOrAdd<PEImage>(peImagePointer);
+    }
```

Comment:
I wonder if we're going to have a problem in practice with the cDAC being more eager to read data from the target process up front.  I think the brittle DAC is lazier about following pointers in target memory.  It might be able to emit more info to the debugger before failing.

Reply:
This is a concern for mini and triage dumps only. What we need to do for those is make sure that our routines for ensuring that enough data structures are live are updated so that they make sure to cover the important scenarios needed for examining dumps. Also we will likely need to have the ability for some of the apis to produce incomplete results. The only reason this works in the existing scheme is that there are a selected set of places in the DAC where failures are just ignored. We will eventually need to add the same things to the cDAC, but I don't think we should do that yet.